dask_jobqueue/remote_slurm.py

```diff
@@ -0,0 +1,195 @@
+        self.api_client_session_kwargs = kwargs.pop("api_client_session_kwargs")
+        self.api_url = kwargs.pop("api_url")
+        self.remote_job_extra = (
+            kwargs.pop("remote_job_extra") if "remote_job_extra" in kwargs else {}
+        )
+        super().__init__(*args, **kwargs)
+
+    @contextmanager
+    def job_file(self):
+        """
+        Override the job_file and pass back the job_script. job_file() is used by Job.start and
+         gives output directly to SLURMJob._submit_job().
+        """
+        yield self.job_script()
+
+    @property
+    def _job_configuration(self) -> Dict:
+        configuration = {
+            "environment": {"name": "test"},
+            "tasks": 1,
+            "cpus_per_task": self.job_cpu or self.worker_cores,
+        }
+
+        if self.job_name is not None:
+            configuration["name"] = self.job_name
+        if self.log_directory is not None:
+            configuration[
+                "standard_error"
+            ] = f"{self.log_directory}/{self.job_name or 'worker'}-%%J.err"
+            configuration[
+                "standard_output"
+            ] = f"{self.log_directory}/{self.job_name or 'worker'}-%%J.out"
+        if self.queue is not None:
+            configuration["partition"] = self.queue
+        if self.project is not None:
+            configuration["account"] = self.project
+
+        memory = self.job_mem
+        if self.job_mem is None:
+            memory = slurm_format_bytes_ceil(self.worker_memory)
+        if memory is not None:
+            configuration["memory_per_node"] = memory
+
+        if self.walltime is not None:
+            configuration["time_limit"] = self.walltime
+
+        configuration.update(self.remote_job_extra)
+
+        return configuration
+
+    async def _submit_job(self, script):
+        # https://slurm.schedmd.com/rest_api.html#slurmctldSubmitJob
+        client_session = aiohttp.ClientSession(
+            raise_for_status=True, **self.api_client_session_kwargs
+        )
+        response = await client_session.post(
+            f"{self.api_url}slurm/v0.0.36/job/submit",
+            json={"script": script, "job": self._job_configuration},
+            raise_for_status=False,
+        )
+        async with response:
+            return await response.json()
+
+    def _job_id_from_submit_output(self, out):
+        # out is the JSON output from _submit_job request.post
+        # See https://slurm.schedmd.com/rest_api.html#v0.0.36_job_submission_response
+        return out["job_id"]
+
+    async def close(self):
+        logger.debug("Stopping worker: %s job: %s", self.name, self.job_id)
+        # https://slurm.schedmd.com/rest_api.html#slurmctldCancelJob
+        client_session = aiohttp.ClientSession(
+            raise_for_status=True, **self.api_client_session_kwargs
+        )
+        response = await client_session.delete(
+            f"{self.api_url}slurm/v0.0.36/job/{self.job_id}",
```

Comment:
Aren't there cases in which someone could be running a different version other than `v0.0.36`? So, should we be hard coding this here?

Reply:
Theoretically yes, if we want to open it up like that then we should probably have users include slurm/v0.0.36 into the api_url itself.

In reality slurmrestd's implementations are nearly identical between 0.0.35/36/37

Reply:
Sounds good... Thank you for the clarification

Reply:
Is this really resolved? I'd say we need a way to configure the version in the URL for future compatibility?